File: backend/src/main/kotlin/com/plugga/backend/service/SideService.java
package com.plugga.backend.service;

import com.plugga.backend.entity.Side;

import java.util.List;

public interface SideService {
    List<Side> findAll();

    Side findById(int id);

    List<Side> findByCardId(int cardId);

    void save(Side side);

    void deleteById(int id);
}

File: backend/src/main/kotlin/com/plugga/backend/service/SideService.kt
package com.plugga.backend.service

import com.plugga.backend.entity.Side
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface SideService {
    fun findAll(pageable: Pageable): Page<Side>

    fun findById(id: Int): Side?

    fun findByCardId(pageable: Pageable, cardId: Int): Page<Side>

    fun save(side: Side)

    fun deleteById(id: Int)
}
